### PR TITLE
split dark/light modes

### LINKED ISSUE #6: customize dark style too

### LINKED ISSUE #36: Review Pat
@maelle Thanks for your work so far! I've noted down some of my observations while looking at the new site. They are mainly cosmetic details, and I am not responsible for these (@christophsax is) but maybe they are still helpful...

## ToDos

- [ ] Align navbar height in dark and light theme
- [ ] Align container width of post preview's between dark and light theme
- [ ] Align button colors (e.g. page buttons) of both dark and light theme with cynkraweb colors
- [x] Add a workflow for previews in PRs (copy from cynkraweb)
- [ ] Dark theme: navbar should also be dark
- [ ] Align link color in posts with blue used in cynkraweb
- [x] Don't linebreak "Except where otherwise noted, content on this blog is licensed under the [CC BY-NC-ND 4.0](https://creativecommons.org/licenses/by-nc-nd/4.0/) license."
- [x] Align capitalization in navbar dropdowns with cynkraweb.
- [ ] Can we link the author names in posts to https://cynkra.com/about?

## Further questions
- Do we want the "reader mode toggle" switch?
- how to stay in sync with changes in cynkraweb? Can we use the cynkraweb navbar in cynkrablog dynamically?

### PR DESCRIPTION
**Pull request by maelle**
Fix #36 
Fix #6 

TODOS 

- [x] Make the navbar appear again in light mode!
- [ ] Fix toggle and search icon in light mode https://github.com/quarto-dev/quarto-cli/discussions/8380
- [x] Find a way to make logo dark. SVG? https://github.com/quarto-dev/quarto-cli/discussions/8379 -- with the current answer to me it seems easier to store an SVG in this repo and change colors based on the theme, I'll do that. Update: this did not work as CSS rules don't apply to the included SVG.

### CODE REVIEW

**Comment by maelle** [comment]
I'm a bit stuck here.

@christophsax who could help with 
- making the text appear in the light mode, I can't find the faulty CSS rule so would appreciate a second pair of eyes.
- writing javascript to switch the logo image to another one for the dark theme.

**Comment by christophsax** [comment]
@DivadNojnarg, perhaps you could help here?

**Comment by maelle** [comment]
@DivadNojnarg at the moment for the light theme we use https://cynkra.com/assets/img/logo_small.svg

Docs of dark theme customization https://quarto.org/docs/output-formats/html-themes.html#dark-mode

Question on Quarto forum https://github.com/quarto-dev/quarto-cli/discussions/8379

**Comment by maelle** [comment]
Regarding the dark mode toggle, currently invisible in light mode. Using the web dev console I see that it responds to "dark" theme things whereas I am looking at it in light mode?!

image

**Comment by maelle** [comment]
@DivadNojnarg for info https://github.com/quarto-dev/quarto-cli/discussions/8601

**Comment by DivadNojnarg** [comment]
@maelle. I am currently checking the branch. Is this expected that the card border color is black in dark mode (we don't see it)?

File: custom.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+// Description: Change image src depending on body class (quarto-light or quarto-dark)
+// License: MIT
+function updateImageSrc() {
+    var bodyClass = window.document.body.classList;
+    var images = window.document.getElementsByTagName('img');
+    for (var i = 0; i < images.length; i++) {
+      var image = images[i];
+      var src = image.src;
+      var newSrc = src;
+      if (bodyClass.contains('quarto-light') && src.includes('_dark')) {
```
Comment (DivadNojnarg) [inline custom.js:12]: @maelle. I just changed the names here because I am not using the same images names.

File: _quarto.yml [hunk 2]
```diff
@@ -1,6 +1,10 @@
 project:
   type: website
   output-dir: docs
+  # For some reasons the logo light isn't properly copied
+  # so we have to do it post-render.
+  post-render: 
+    - "cp ./logo_light.svg docs/"
```
Comment (DivadNojnarg) [inline _quarto.yml:7]: If there's a better way to quarto to copy tell logo_light.svg. Using `resources` does not work: https://quarto.org/docs/websites/website-tools.html#site-resources. At least that works.

**Comment by DivadNojnarg** [comment]
What still remain is the look of the navbar dropdown elements to make them in line with cynkra.com (rounded border, hover background color, ...)

Screenshot 2024-02-08 at 11 40 08

**Comment by maelle** [comment]
@DivadNojnarg regarding the card border, I might have been badly imitating https://cynkra.com/consulting/ where there are no borders, but it looks better with borders indeed.

File: custom.js (hunk 1 above)
Comment (maelle) [reply custom.js:12]: ok!

File: _quarto.yml (hunk 2 above)
Comment (maelle) [reply _quarto.yml:7]: I think this is smart!

**Comment by maelle** [comment]
The logo switch is beautiful :star_struck:

**Comment by maelle** [comment]
waiting for an answer on https://github.com/quarto-dev/quarto-cli/discussions/8601 before I tweak further

**Pull request merged**